### ISSUE TITLE
dev branch SDK url changed

### ISSUE THREAD

**Issue by enricobenedos** [aloisdeniel/vsts-flutter-tasks#46]
It seems that from `v1.17.0-dev0.0` Flutter team has changed the SDK download url format for dev branch.
First:
`https://storage.googleapis.com/flutter_infra/releases/dev/macos/flutter_macos_v1.16.3-dev.zip`
Now
`https://storage.googleapis.com/flutter_infra/releases/dev/macos/flutter_macos_1.17.0-dev.0.0-dev.zip`

It seems that they remove the `v` letter from the url. Actually if I configure pipeline to work with dev branch this error is returned
```console
Starting: FlutterInstall
==============================================================================
Task         : Flutter Install
Description  : Install the Flutter environment.
Version      : 0.2.15
Author       : Aloïs Deniel
Help         : For more information, take a look at the Flutter [documentation](https://flutter.io)
==============================================================================
(node:990) Warning: Use Cipheriv for counter mode of aes-256-ctr
(node:990) Warning: Use Cipheriv for counter mode of aes-256-ctr
(node:990) Warning: Use Cipheriv for counter mode of aes-256-ctr
(node:990) Warning: Use Cipheriv for counter mode of aes-256-ctr
(node:990) Warning: Use Cipheriv for counter mode of aes-256-ctr
(node:990) Warning: Use Cipheriv for counter mode of aes-256-ctr
Downloading: https://storage.googleapis.com/flutter_infra/releases/dev/macos/flutter_macos_v.18.0-dev.4.0-dev.zip
##[error]Error: Unexpected HTTP response: 404
```
Can it be an idea download SDK trying first with an url version and then with another one (first with `v` and then without `v`)?

**Comment by cgravato** [comment]
It's also happening with the Beta channel.

**Comment by jlalvarez18** [comment]
Please fix this.

**Comment by mikeesouth** [comment]
I also have this problem. I submitted a PR for this. Can this be approved asap?
https://github.com/aloisdeniel/vsts-flutter-tasks/pull/47

Ping @aloisdeniel

**Comment by FifoTheHein** [comment]
Also stuck with this issue. Any feedback @aloisdeniel ?

**Comment by JordanADavies** [comment]
Also happening on stable now.

**Comment by aloisdeniel** [comment]
I was waiting for the stable release. I will update the extension!

**Comment by mikeesouth** [comment]
@aloisdeniel note that I have a PR that fixes this but it's a crude fix. A better fix could be to use the archive path in the releases json file. I also noticed that currently, the stable and beta branch points to the same hash which also breaks this plugin a bit (even after the v prefix fix).
From the release json file here: https://storage.googleapis.com/flutter_infra/releases/releases_windows.json we can see the current release hashes:
"current_release": {
    "beta": "e6b34c2b5c96bb95325269a29a84e83ed8909b5f",
    "dev": "c2b7342ca470b11cfaad4fbfb094f73aa4c85320",
    "stable": "e6b34c2b5c96bb95325269a29a84e83ed8909b5f"
  },

Note that beta and stable both points to the same hash.

The stable entry looks like this:
{
      "hash": "e6b34c2b5c96bb95325269a29a84e83ed8909b5f",
      "channel": "stable",
      "version": "1.17.0",
      "release_date": "2020-05-06T14:30:41.659932Z",
      "archive": "stable/windows/flutter_windows_1.17.0-stable.zip",
      "sha256": "b6e8ca43b046d1a10cfb995b8c63828c8fbe35bb55e9c58349982810d57b2681"
    },

The beta entry looks like this:
{
      "hash": "e6b34c2b5c96bb95325269a29a84e83ed8909b5f",
      "channel": "beta",
      "version": "1.17.0-3.4.pre",
      "release_date": "2020-05-02T20:33:14.297290Z",
      "archive": "beta/windows/flutter_windows_1.17.0-3.4.pre-beta.zip",
      "sha256": "9120c5044fbbec16b639b2a5256f37fd0774e976798ff5aa44b9ecbc4c173152"
    },

I'm using the beta branch and the current plugin first gets the beta hash but then it looks up the first hit on that hash (which is the stable entry) and then it tries to concatenate a URL with beta in the url and beta as suffix on the zip file - but with the version string from the stable entry. This results in a 404.

This can be circumvented by changing to the stable release but still, maybe something to look into if you fix this.

**Comment by hey24sheep** [comment]
Have a look at this for installation implementation [link](https://github.com/cross-solutions/install-flutter-sdk/blob/master/src/install/index.ts), Also @aloisdeniel your builds are failing and we need this update, could you please look into it.

**Comment by hey24sheep** [comment]
I have fixed and published it Check my fork here repo and here is the extension. Any more feature and prs are welcome on my fork.

**Comment by aloisdeniel** [comment]
It should be fixed now with latest version (0.2.30).

Let me know if there's still issues on your side.

**Comment by mkieres** [comment]
@aloisdeniel Thanks a bunch for you prompt reaction and the fix! I can confirm that my build is now passing.
